trillian_log_api.proto

```diff
@@ -55,6 +55,12 @@ message LogLeaf {
     // If leaf_identity_hash is empty, it's assumed to be the same as the
     // merkle_leaf_hash.
     bytes leaf_identity_hash = 5;
+    // queue_timestamp_nanos records the time at which this leaf was first passed to QueueLeaves.
+    // This value will be determined and set by the LogServer.
+    int64 queue_timestamp_nanos = 6;
```

Comment:
Prefer using google.protobuf.Timestamp to represent time.

(`import "google/protobuf/timestamp.proto"` / [link](https://github.com/google/protobuf/blob/master/src/google/protobuf/timestamp.proto#L43))

Reply:
Could you outline the benefit?
We're storing nanos since epoch, which with 63 bits is fine up until 2262-04-11 23:47:16, but using an embedded timestamp message has the cost of extra complexity if/when we decide we want to include the `QueueTimestampNanos` under the SET signature.

Reply:
OK, I'll bite. A few benefits are:

* Using a proper type to represent time instead of a primitive, which:
    * Encourages callers to handle it correctly
    * Removes ambiguity in regards to time unit
    * Just all-around good design advice
* Consistency with our own APIs ([link](https://github.com/google/trillian/blob/3f6222aa42face2c2d81a24de5438d95e4f4bb3d/trillian.proto#L161))
* Consistency with the Google API Style ([link](https://cloud.google.com/apis/design/standard_fields))

You can still get the nanos for the SET, so that's not really an issue. If you want to go down the "using a proper type to represent time adds too much complexity" route I won't try to stray you away from it. I can assure you, though, that an API reviewer won't buy this either.

Reply:
On consistency, I'll point out [this](https://github.com/google/trillian/blob/master/trillian.proto#L187) which existed _long_ before `Tree.CreateTime` :)

> You can still get the nanos for the SET, so that's not really an issue. 

Well, there _is_ one issue: 63 bits of seconds + 31 bits of nanoseconds can represent a much _much_ bigger timestamp than 63 bits of nanos can, so we'll have to be very careful about what we define the serialization format for SET and STH to be if we were ever going to use `Timestamp` anywhere other than as a read-only representation...

Reply:
Were you aware of Timestamp when you defined SignedLogRoot? I'd be happy to update it to use Timestamp, but I figured that might get strong push back. 

Re Timestamp's precision, although the type definition allows for a "larger" representation, the defining factor is still storage, so I don't think that changes much. We don't have to use all the precision it allows for. Plus, nanos-since-epoch should be good until "2262-04-11 23:47:16", right?